### PR TITLE
Common beam and convolution kernel docs

### PR DESCRIPTION
**Pull request by e-koch**
* [x] Explain how the common beam is found in radio-beam, and to try when the deconvolution error
* [x] How to create kernel for convolution (Gaussian and top-hat)

### CODE REVIEW

File: docs/commonbeam.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
+.. _com_beam:
+
+Finding the smallest common beam
+================================
+
+radio-beam implements an exact solution for sets of 2 beams and an approximate method---`the Khachiyan algorithm <https://en.wikipedia.org/wiki/Ellipsoid_method>`_---for larger sets. The former case is straightforward to compute as the beams can be transformed into a space where the larger beam is circular to find the overlap area (`~radio_beam.commonbeam.common_2beams`). Our implementation borrows from the implementation in `CASA <https://casa.nrao.edu/>`_ (see `here <https://open-bitbucket.nrao.edu/projects/CASA/repos/casa/browse/code/imageanalysis/ImageAnalysis/CasaImageBeamSet.cc>`_).
```
Comment (keflavich) [inline docs/commonbeam.rst]: Can you comment on where the known failures in CASA are?  It does fail in some cases, right?

**Comment by e-koch** [comment]
@keflavich -- mind taking a look at the convolution kernel page too?

File: docs/convolution_kernels.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Top-Hat
+^^^^^^^
+
+`~radio_beam.Beam.as_tophat_kernel` returns an elliptical top-hat kernel scales to have the same area as a Gaussian kernel within the FWHM.  Similar to the Gaussian kernel, only the pixel scale needs to be given::
+
+    >>> tophat_kern = my_beam.as_tophat_kernel(pix_scale)
+
+`tophat_kern` is a `~radio_beam.beam.EllipticalTophat2DKernel` object, also derived from `Kernel2D <http://docs.astropy.org/en/stable/api/astropy.convolution.Kernel2D.html#astropy.convolution.Kernel2D>`__ in astropy's convolution package. Keyword arguments can be passed to `~radio_beam.Beam.as_tophat_kernel`.
+
+The values in the kernel are normalized to unity and is suitable for convolution.  However, the top-hat kernel is also useful for masking purposes, in which case a boolean version of the kernel is useful.  To make a boolean version, we need to access the array in the kernel object and look for non-zero values::
```
Comment (keflavich) [inline docs/convolution_kernels.rst]: grammar correction.  
```suggestion
The values in the kernel are normalized to unity, and it is suitable for convolution.  However, the top-hat kernel is also useful for masking purposes, in which case a boolean version of the kernel is useful.  To make a boolean version, we need to access the array in the kernel object and look for non-zero values::
```

File: docs/convolution_kernels.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+From a `~radio_beam.Beams` object, a convolution kernel can be made for each beam in the set by slicing::
+
+    >>> from radio_beam import Beams
+    >>> from astropy.io import fits
+    >>> bin_hdu = fits.open('file.fits')[1]  # doctest: +SKIP
+    >>> beams = Beams.from_fits_bintable(bin_hdu)  # doctest: +SKIP
+    >>> beams[0].as_kernel(pix_scale)  # doctest: +SKIP
+
+Kernels can be created for each individual beam in the set.
```
Comment (keflavich) [inline docs/convolution_kernels.rst]: Remove or modify this last sentence; it's redundant with the overview description
```suggestion
```

File: docs/commonbeam.rst (hunk 1 above)
Comment (e-koch) [reply docs/commonbeam.rst]: :+1:

**Pull request merged**